### PR TITLE
core: Rename register_* with load_*

### PR DESCRIPTION
**Pull request by math-fehr**
This PR renames `register_*` with `load_*`.
`load_*` is the naming used by MLIR, so it makes sense to go there.
Also, I would like to add a `register_*` methods, as in #1116

### CODE REVIEW

File: docs/irdl.ipynb [hunk 1]
```diff
@@ -1518,8 +1518,14 @@
...
  "metadata": {
+  "kernelspec": {
+   "display_name": "venv",
+   "language": "python",
+   "name": "python3"
+  },
   "language_info": {
-   "name": "python"
+   "name": "python",
+   "version": "3.10.12"
```
Comment (superlopuh) [inline docs/irdl.ipynb:1528]: .

File: docs/tutorial.ipynb [hunk 2]
```diff
@@ -1,501 +1,501 @@
...
- ],
- "metadata": {
-  "language_info": {
-   "name": "python"
-  }
- },
- "nbformat": 4,
- "nbformat_minor": 5
-}
+    "cells": [
```
Comment (superlopuh) [inline docs/tutorial.ipynb:2]: 😱

**Review by superlopuh — CHANGES_REQUESTED** [review]
couple of stray diffs

**Comment by math-fehr** [comment]
> couple of stray diffs

Oops, that should be fixed now, thanks!

**Review by superlopuh — APPROVED** [review]

**Review by webmiche — APPROVED** [review]

**Pull request merged**